### DISCUSSION TITLE
Conversion from SimpleITK image to torchio Image

### DISCUSSION THREAD

**Discussion by mattwarkentin** [TorchIO-project/torchio#612] [Q&A] [answered]
Hi,

I know you can convert from a `torchio` Image to a SimpleITK image using the `image.as_sitk()` method. But is there an efficient way to go from a SimpleITK image back to `torchio`? Something like `from_sitk()`?

**Comment by mattwarkentin** [comment, marked as answer]
It seems to me that since `torchio` supports reading in any image from disk that is supported by SimpleITK or nibabel, that it should also support creating a `torchio` `Image` using in-memory SimpleITK or nibabel images. What do you think?

**Reply by fepegar** [reply]
Hi, @mattwarkentin. I think that's a good idea. We could add a method to `Image`, for example:

```python
@classmethod
def from_sitk(cls, sitk_image):
    tensor, affine = sitk_to_nib(sitk_image)
    return cls(tensor=tensor, affine=affine)
```

Where `sitk_to_nib` would need to be imported.

Then, you could do, for example:

```python
tio.ScalarImage.from_sitk(sitk_image)
```

Would you like to contribute with a PR?

**Reply by mattwarkentin** [reply]
I agree with the approach. Sure, I would be happy to contribute a PR to this project.

**Comment by fepegar** [comment]
Great, thanks!
